src/app/researcher/study/[encodedStudyId]/edit/form.tsx

```diff
@@ -1,84 +1,130 @@
 'use client'
 
-import React from 'react'
+import React, { useState } from 'react'
 
 import { Form as HookForm, useForm } from 'react-hook-form'
-import { Checkbox, Select, Textarea, TextInput } from 'react-hook-form-mantine'
-
+import { Checkbox, Textarea, TextInput } from 'react-hook-form-mantine'
 import { inputStyle, labelStyle } from './style.css'
-import { Button, Flex, Stack, Text } from '@mantine/core'
+import { Button, Flex, Group, Stack, Text } from '@mantine/core'
 import { onUpdateStudyAction } from './actions'
 import { FormValues, schema, zodResolver } from './schema'
+import Link from 'next/link'
 
 export const Form: React.FC<{ studyId: string; study: FormValues }> = ({ studyId, study }) => {
-    const { control } = useForm<FormValues>({
+    const {
+        control,
+        // formState: { isValid },
+        getValues,
+    } = useForm<FormValues>({
         resolver: zodResolver(schema),
```

Comment:
This is a nitpick but i prefer having the schema defined  in the same file (its just easier for me to read) but no changes here necessary if your team prefers otherwise

Reply:
putting the schema in a separate file lets us re-use it for server-side validation in the `actions.ts` file to validate the incoming values before using them with the DB.    It also can generate the typescript types for the form and that can be imported separately as well